galaxy_importer/schema.py

```diff
@@ -375,6 +376,138 @@ def parse(cls, data):
+
+    @role_name.validator
+    def _validate_role_name(self, attribute, value):
+        """Ensure role name matches the regular expression."""
+
+        if value is not None and not constants.NAME_REGEXP.match(value):
+            raise ValueError(f"role name {value} is invalid")
+
+    @author.validator
+    def _validate_author(self, attribute, value):
+        """Ensure the author value is not too long."""
+
+        if value is not None and len(value) > MAX_LENGTH_AUTHOR:
+            raise ValueError(f"author must not exceed {MAX_LENGTH_AUTHOR} characters")
+
+    @issue_tracker_url.validator
+    def _validate_url(self, attribute, value):
+        """Ensure a URL is not too long."""
+
+        if value is not None and len(value) > MAX_LENGTH_URL:
+            raise ValueError(f"url must not exceed {MAX_LENGTH_URL} characters")
+
+    @license.validator
+    def _validate_license(self, attribute, value):
+        """Ensure the license is not too long."""
+
+        if value is not None:
+            if len(value) > MAX_LENGTH_LICENSE * 2:
+                raise ValueError(
+                    f"role license must not exceed {MAX_LENGTH_LICENSE * 2} characters"
+                )
+
+    @min_ansible_version.validator
+    @min_ansible_container_version.validator
+    def _validate_version(self, attribute, value):
+        """Ensure a version is not too long and abides by semantic versioning."""
+
+        if value is not None and len(str(value)) > MAX_LENGTH_VERSION:
+            raise ValueError(
+                f"version for {attribute.name} must not exceed {MAX_LENGTH_VERSION} characters"
+            )
+
+    @galaxy_tags.validator
+    def _validate_tags(self, attribute, value):
+        """Ensure tags are not too long."""
+
+        if value is not None:
+            if any(len(element) > MAX_LENGTH_TAG for element in value):
+                raise ValueError(f"tag must not exceed {MAX_LENGTH_TAG} characters")
+
+
+@attr.s(frozen=True)
+class LegacyMetadata:
+    """Represents legacy role metadata."""
+
+    galaxy_info = attr.ib(default=None, type=LegacyGalaxyInfo)
+    dependencies = attr.ib(factory=list)
+
+    @classmethod
+    def parse(cls, data):
+        with open(data, "r") as fh:
+            metadata = yaml.safe_load(fh)
+            if not isinstance(metadata, dict):
+                raise ValueError("metadata must be in the form of a yaml dictionary")
+            if "galaxy_info" not in metadata:
+                raise ValueError("galaxy_info field not found in metadata")
+            if not isinstance(metadata["galaxy_info"], dict):
+                raise ValueError("galaxy_info field must contain a dictionary")
+            try:
+                galaxy_info = LegacyGalaxyInfo(**metadata["galaxy_info"])
+            except TypeError as e:
+                raise ValueError("unknown field in galaxy_info") from e
+            dependencies = metadata.get("dependencies", list())
```

Comment:
does this line effectively do this?
```
dependencies = metadata.get("dependencies")
dependencies = list(dependencies)
```

Reply:
No, if there is no "dependencies" field in metadata, then `metadata.get("dependencies")` returns `None`. The subsequent call would then be `list(None)` which results in the runtime error `TypeError: 'NoneType' object is not iterable`

Reply:
The `list()` in my implementation replaces `None` as the default value for if there is no "dependencies" field in metadata, ensuring that dependencies will be a list